backend/core/llm_chain.py

```
@@ -3,490 +3,167 @@
-        logger.error(f"Failed to create retrieval chain: {e}")
-        raise
-
+def create_multi_vector_retriever(docs, embeddings):
+    vectorstores = {}
+    docs_by_source = {
+        "resume": [doc for doc in docs if doc.metadata["source"] == "resume"],
+        "about": [doc for doc in docs if doc.metadata["source"] == "about"],
+        "illustration": [doc for doc in docs if doc.metadata["source"] == "illustration"],
+    }
+    for source, source_docs in docs_by_source.items():
+        if not source_docs: continue
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)